reframe/core/pipeline.py

```diff
@@ -1092,6 +1092,12 @@ def setup(self, partition, environ, **job_opts):
               <migration_2_to_3.html#force-override-a-pipeline-method>`__ for
               more details.
 
+           .. versionchanged:: 3.4
+              Overriding this method directly unless you are in
+              special test. See `here
+              <migration_2_to_3.html#force-override-a-pipeline-method>`__ for
+              more details.
```

Comment:
I don't think that this adds anything. I would suggest to remove. Just add an "a" here: "unless you are in a special test."